comp/dogstatsd/server/enrich.go

```diff
@@ -62,9 +67,8 @@ func extractTagsMetadata(tags []string, originFromUDS string, processID uint32,
 		} else if strings.HasPrefix(tag, CardinalityTagPrefix) && origin.Cardinality == "" {
 			origin.Cardinality = tag[len(CardinalityTagPrefix):]
 			continue
-		} else if strings.HasPrefix(tag, jmxCheckNamePrefix) {
-			checkName := tag[len(jmxCheckNamePrefix):]
-			metricSource = metrics.JMXCheckNameToMetricSource(checkName)
+		} else if newSource := getMetricSourceFromTag(metricSource, tag); newSource != metricSource {
```

Comment:
`newSource != metricSource` means that non-duplicate tags with the same prefix would not be processed and would remain attached to the metric. Is there any reason to do so?

This is also different to how jmxfetch tag was handled before.

Reply:
Oof I missed the fact we are extracting the found tag. The new vs old source was an attempt to fix some overwriting, but this is not necessary. I will redo this to be more serverless specific and leave the JMX logic untouched